lib/model/folder_sendrecv.go

```diff
@@ -1458,7 +1466,17 @@ func (f *sendReceiveFolder) pullBlock(state pullBlockState, out chan<- *sharedPu
 		// leastBusy can select another device when someone else asks.
 		activity.using(selected)
 		var buf []byte
-		buf, lastError = f.model.requestGlobal(selected.ID, f.folderID, state.file.Name, state.block.Offset, int(state.block.Size), state.block.Hash, state.block.WeakHash, selected.FromTemporary)
+		done := make(chan struct{})
+		go func() {
+			buf, lastError = f.model.requestGlobal(selected.ID, f.folderID, state.file.Name, state.block.Offset, int(state.block.Size), state.block.Hash, state.block.WeakHash, selected.FromTemporary)
+			close(done)
+		}()
+		select {
+		case <-done:
+		case <-f.ctx.Done():
+			state.fail(errors.Wrap(f.ctx.Err(), "folder stopped"))
+			return
```

Comment:
This actually changes behaviour by exiting the loop early, not passing stuff to finisher to unregister for progress emission etc.

Also, what does this change have to do with the spirit of this PR?